### PR TITLE
Router perf v2

### PR DESCRIPTION
**Pull request by rsevilla87**
Re-implementation of the router scalability test suite.

Example documents at:
https://search-perfscale-dev-chmf5l4sh66lvxbnadi4bznl3a.us-west-2.es.amazonaws.com/_plugin/kibana/goto/225d43996365ad560e7958378b8d11af

To execute it just issue:
```
./ingress-performance.sh
```

The workload customizations are well described in the docs

Signed-off-by: Raul Sevilla <rsevilla@redhat.com>

### CODE REVIEW

File: workloads/router-perf-v2/common.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    for sample in ${SAMPLES}; do
+      log "Executing sample ${sample} from termination ${TERMINATION} with ${N_CLIENTS} clients and ${N_KEEPALIVE_REQUESTS} keepalive requests"
+      kubectl exec -n http-scale-client -it ${client_pod} -- python3 /workload/workload.py --mb-config /tmp/mb-${TERMINATION}.json --termination ${TERMINATION} --runtime ${RUNTIME} --ramp-up ${RAMP_UP} --output /tmp/results.csv --sample ${sample}
+      log "Sleeping for ${QUIET_PERIOD} before next test"
+      sleep ${QUIET_PERIOD}
+    done
+  done
+}
+
+enable_ingress_opreator(){
```
Comment (chaitanyaenr) [inline workloads/router-perf-v2/common.sh:70]: s/opreator/operator

File: workloads/router-perf-v2/common.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  kubectl create configmap -n http-scale-client workload --from-file=workload.py
+  log "Adding workload.py to the client pod"
+  oc set volumes -n http-scale-client deploy/http-scale-client --type=configmap --mount-path=/workload --configmap-name=workload --add
+  kubectl rollout status -n http-scale-client deploy/http-scale-client
+  client_pod=$(kubectl get pod -l app=http-scale-client -n http-scale-client | grep Running | awk '{print $1}')
+}
+
+tune_liveness_probe(){
+  log "Disabling cluster version and ingress operators"
+  kubectl scale --replicas=0 -n openshift-cluster-version deploy/cluster-version-operator
```
Comment (chaitanyaenr) [inline workloads/router-perf-v2/common.sh:42]: Maybe use one of the clients - either kubectl or oc for the operations to be consistent? Thoughts?

Comment (rsevilla87) [reply workloads/router-perf-v2/common.sh:42]: yeah, makes sense, changing it.

File: workloads/router-perf-v2/common.sh (hunk 1 above)
Comment (rsevilla87) [reply workloads/router-perf-v2/common.sh:70]: good catch!

**Comment by Harshith-umesh** [comment]
Results for e2e-benchmarking CI Tests
Workload                | Test                           | Result | Runtime  |
------------------------|--------------------------------|--------|----------|
etcd-perf | run_etcd_tests_fromgit.sh | PASS | 00:02:35
scale-perf | run_scale_fromgit.sh | FAIL | 00:15:00
network-perf | smoke_test.sh | PASS | 00:09:13
kube-burner | run_clusterdensity_test_fromgit.sh | PASS | 00:04:39
kube-burner | run_maxnamespaces_test_fromgit.sh | PASS | 00:04:04
kube-burner | run_maxservices_test_fromgit.sh | PASS | 00:04:06
kube-burner | run_nodedensity_test_fromgit.sh | FAIL | 00:00:04
kube-burner | run_nodedensity-heavy_test_fromgit.sh | FAIL | 00:00:04
storage-perf | run_storage_tests_fromgit.sh | PASS | 00:04:34
upgrade-perf | run_upgrade_fromgit.sh | PASS | 00:00:22
router-perf | run_router_test.sh | FAIL | 00:00:12

File: workloads/router-perf-v2/templates/http-scale-client.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        - name: UUID
+          value: {{.UUID}}
+        - name: HOST_NETWORK
+          value: "{{.hostNetwork}}"
+        - name: NUMBER_OF_ROUTERS
+          value: "{{.numberOfRouters}}"
+        command:
+        - sleep
+        - inf
+        image: quay.io/rsevilla/http-stress:latest
```
Comment (rsevilla87) [inline workloads/router-perf-v2/templates/http-scale-client.yml:37]: Pending to update the image location

**Comment by Harshith-umesh** [comment]
@rsevilla87 there are problems in the scale-perf and router-perf workloads rn , they don't take into considerations clusters which don't have a workload node on them such as the one which is being used for the e2e-CI, that's why both of those are failing and the kube-burner node density and heavy-density are failing coz number of pods to start < number which are already present , fixing scale-perf should fix these as well .

**Comment by Harshith-umesh** [comment]
> @rsevilla87 there are problems in the scale-perf and router-perf workloads rn , they don't take into considerations clusters which don't have a workload node on them such as the one which is being used for the e2e-CI, that's why both of those are failing and the kube-burner node density and heavy-density are failing coz number of pods to start < number which are already present , fixing scale-perf should fix these as well .

**Comment by dry923** [comment]
@rsevilla87 Can you add an allowable taints option? Given we are allowing the user to pick a workload node they may have an issue if its tainted.

**Comment by rsevilla87** [comment]
> @rsevilla87 Can you add an allowable taints option? Given we are allowing the user to pick a workload node they may have an issue if its tainted.

Hey Russ, the client pod has tolerations to support the workload taints, just fixed an indentation issue in https://github.com/cloud-bulldozer/e2e-benchmarking/pull/124/commits/4c10ac31674a17c43244e5e75d85127dd4b5996e that screwed my demo :)

File: ci/env.sh [hunk 4]
```diff
@@ -27,5 +28,15 @@ export PBENCH_SERVER='pbench.dev.openshift.com'
...
+export HTTP_TEST_ROUTE_TERMINATION='http edge mix'
 export HTTP_TEST_RUNTIME=10
+
+# For router perf v2
+export NUMBER_OF_ROUTES=10
+export RUNTIME=10
+export SAMPLES=2
+export TERMINATIONS=http
+export KEEPALIVE_REQUESTS="1 40"
+export CLIENTS="1 10"
```
Comment (mohit-sheth) [inline ci/env.sh:40]:
```
export KEEPALIVE_REQUESTS="1 10"
￼export CLIENTS="1 40"
```

**Review by mohit-sheth — APPROVED** [review]
LGTM, nice work @rsevilla87

File: ci/CI_test_list.yml [hunk 5]
```diff
@@ -9,5 +10,3 @@ kube-burner:run_nodedensity-heavy_test_fromgit.sh
 storage-perf:run_storage_tests_fromgit.sh 
 upgrade-perf:run_upgrade_fromgit.sh 
 router-perf:run_router_test.sh
```
Comment (chaitanyaenr) [inline ci/CI_test_list.yml:12]: @rsevilla87 thoughts on removing the router-perf workload from the repo and replacing it with the v2?

Comment (rsevilla87) [reply ci/CI_test_list.yml:12]: That's the idea, we can remove/replace it in another PR. (Didn't want to make any disrupting changes in this one)

Comment (chaitanyaenr) [reply ci/CI_test_list.yml:12]: Sounds good.

**Review by chaitanyaenr — APPROVED** [review]
LGTM

**Pull request merged**